### PR TITLE
[LesliesPoolmart US] Fix and rename spider

### PR DESCRIPTION
**Pull request by Cj-Malone**
```python
{"atp/brand/Leslie's Poolmart": 931,
 'atp/brand_wikidata/Q6530568': 931,
 'atp/category/shop/swimming_pool': 931,
 'atp/cdn/cloudflare/response_count': 938,
 'atp/cdn/cloudflare/response_status_count/200': 938,
 'atp/field/city/missing': 931,
 'atp/field/country/from_spider_name': 931,
 'atp/field/email/missing': 931,
 'atp/field/image/missing': 931,
 'atp/field/lat/missing': 1,
 'atp/field/lon/missing': 1,
 'atp/field/opening_hours/missing': 1,
 'atp/field/operator/missing': 931,
 'atp/field/operator_wikidata/missing': 931,
 'atp/field/phone/missing': 1,
 'atp/field/postcode/missing': 931,
 'atp/field/state/from_reverse_geocoding': 930,
 'atp/field/state/missing': 1,
 'atp/field/street_address/missing': 931,
 'atp/field/twitter/missing': 931,
 'atp/nsi/perfect_match': 931,
 'downloader/request_bytes': 739236,
 'downloader/request_count': 938,
 'downloader/request_method_count/GET': 938,
 'downloader/response_bytes': 26853703,
 'downloader/response_count': 938,
 'downloader/response_status_count/200': 938,
 'elapsed_time_seconds': 8.559062,
 'feedexport/success_count/FileFeedStorage': 1,
 'finish_reason': 'finished',
 'finish_time': datetime.datetime(2024, 1, 30, 17, 10, 22, 461990, tzinfo=datetime.timezone.utc),
 'httpcache/hit': 938,
 'httpcompression/response_bytes': 162860571,
 'httpcompression/response_count': 938,
 'item_dropped_count': 1,
 'item_dropped_reasons_count/DropItem': 1,
 'item_scraped_count': 931,
 'log_count/INFO': 10,
 'log_count/WARNING': 1,
 'memusage/max': 180350976,
 'memusage/startup': 180350976,
 'request_depth_max': 3,
 'response_received_count': 938,
 'robotstxt/request_count': 1,
 'robotstxt/response_count': 1,
 'robotstxt/response_status_count/200': 1,
 'scheduler/dequeued': 937,
 'scheduler/dequeued/memory': 937,
 'scheduler/enqueued': 937,
 'scheduler/enqueued/memory': 937,
 'start_time': datetime.datetime(2024, 1, 30, 17, 10, 13, 902928, tzinfo=datetime.timezone.utc)}
```

### CODE REVIEW

File: locations/spiders/leslies_poolmart_us.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    name = "leslies_poolmart_us"
+    item_attributes = {"brand": "Leslie's Poolmart", "brand_wikidata": "Q6530568"}
+    sitemap_urls = ["https://lesliespool.com/robots.txt"]
+    sitemap_rules = [("/location/", "parse")]
+
+    def parse(self, response: Response, **kwargs: Any) -> Any:
+        item = Feature()
+        item["ref"] = response.xpath("////a[@data-store-id]/@data-store-id").get()
+        item["lat"] = response.xpath("//@data-store-latitude").get()
+        item["lon"] = response.xpath("//@data-store-longitude").get()
```
Comment (iandees) [inline locations/spiders/leslies_poolmart_us.py:22]: All the lat/lons seem to be set to the same point in the map. Is this pulling the wrong location somehow?

**Comment by Cj-Malone** [comment]
I can't explain this. If we take a look at the first, or any POI, from the [GitHub run](https://alltheplaces-data.openaddresses.io/ci/7715101527/leslies_poolmart_us/output.geojson). But `-87.74069` doesn't show up in the page `https://lesliespool.com/location/saint-george-utah-saint-george-ut-941.html`

**Comment by Cj-Malone** [comment]
It might be a closest store thing, that only gets populated on US requests.

**Comment by iandees** [comment]
Weird, thanks for checking.

**Comment by Cj-Malone** [comment]
Thanks for noticing

**Pull request merged**